tests/cmdline/commands/test_computer.py

```diff
@@ -677,64 +689,107 @@ def test_computer_delete(self):
-        self.assertEqual(self.comp.get_mpirun_command(), new_computer.get_mpirun_command())
-        self.assertEqual(self.comp.get_default_mpiprocs_per_machine(), new_computer.get_default_mpiprocs_per_machine())
-        self.assertEqual(self.comp.get_prepend_text(), new_computer.get_prepend_text())
-        self.assertEqual(self.comp.get_append_text(), new_computer.get_append_text())
-
-    def test_computer_duplicate_non_interactive(self):
-        """Test if 'verdi computer duplicate' in non-interactive mode."""
-        label = 'computer_duplicate_noninteractive'
-        result = self.cli_runner.invoke(
-            computer_duplicate,
-            ['--non-interactive', f'--label={label}', str(self.comp.pk)]
+
+@pytest.fixture(scope='function')
+def comp(temp_dir):
+    """Get an AiiDA computer for localhost.
+
+    Usage::
+
+      def test_1(aiida_localhost):
+          label = aiida_localhost.label
+          # proceed to set up code or use 'aiida_local_code_factory' instead
+
+
+    :return: The computer node
+    :rtype: :py:class:`aiida.orm.Computer`
+    """
+    from aiida.common.exceptions import NotExistent
+    from aiida.orm import Computer
+
+    label = 'computer_duplicate_interactive'
+
+    try:
+        computer = Computer.objects.get(label=label)
+    except NotExistent:
+        computer = Computer(
+            label=label,
+            description='localhost computer set up by test manager',
+            hostname='localhost',
+            workdir=temp_dir,
+            transport_type='core.local',
+            scheduler_type='core.direct'
         )
-        self.assertIsNone(result.exception, result.output)
+        computer.store()
+        computer.set_minimum_job_poll_interval(0.)
+        computer.configure()
 
-        new_computer = orm.Computer.objects.get(label=label)
-        self.assertEqual(self.comp.description, new_computer.description)
-        self.assertEqual(self.comp.hostname, new_computer.hostname)
-        self.assertEqual(self.comp.transport_type, new_computer.transport_type)
-        self.assertEqual(self.comp.scheduler_type, new_computer.scheduler_type)
-        self.assertEqual(self.comp.get_shebang(), new_computer.get_shebang())
-        self.assertEqual(self.comp.get_workdir(), new_computer.get_workdir())
-        self.assertEqual(self.comp.get_mpirun_command(), new_computer.get_mpirun_command())
-        self.assertEqual(self.comp.get_default_mpiprocs_per_machine(), new_computer.get_default_mpiprocs_per_machine())
-        self.assertEqual(self.comp.get_prepend_text(), new_computer.get_prepend_text())
-        self.assertEqual(self.comp.get_append_text(), new_computer.get_append_text())
+    return computer
 
 
+@pytest.mark.usefixtures('clear_database_before_test')
+@pytest.mark.parametrize('non_interactive_editor', ('vim -cwq',), indirect=True)
+def test_computer_duplicate_interactive(run_cli_command, comp, non_interactive_editor):
```

Comment:
Hi @sphuber , thanks for the advice. I add the fixture but seems not work.  Maybe is the problem  is the `user_input` is wrong? Because for the next `test_computer_duplicate_non_interactive` test, I need to pass `--hostname=localhost`, where this `hostname` is a required parameter for the `verdi computer duplicate`.

Reply:
The test that seems to be hanging though is `test_computer_test`. That is part of a class that you haven't converted to `pytest` yet (although you did change it from `self.assertTrue` to `assert`). I am not sure, but maybe the fact that you are mixing two styles is causing the problem?

Reply:
I found the problem and pushed a commit with the fix. See commit message for details